src/utils/__tests__/async-test.js

```diff
@@ -1,14 +1,53 @@
 /* @flow strict-local */
 import { sleep, tryUntilSuccessful } from '../async';
+import { Lolex } from '../../__tests__/aux/lolex';
 
 describe('sleep', () => {
+  const lolex: Lolex = new Lolex();
+
+  afterAll(() => {
+    lolex.dispose();
+  });
+
+  afterEach(() => {
+    // clear any unset timers
+    lolex.clearAllTimers();
+  });
+
   test('waits for a given time in milliseconds', async () => {
     const expectedMs = 1000;
-    const start = Date.now();
-    await sleep(expectedMs);
-    const durationMs = Date.now() - start;
-    expect(expectedMs).toBeLessThanOrEqual(durationMs);
-    expect(durationMs).toBeLessThan(10 * expectedMs);
+
+    const sleepAsync = async () => {
+      const start = Date.now();
+      await sleep(expectedMs);
+      const actualMs = Date.now() - start;
+
+      return actualMs;
+    };
+    const sleepPromise: Promise<number> = sleepAsync();
```

Comment:
I like this simplification, thanks!

Can we make this another step more straightforward from here? I think it'd be helpful to basically inline this single-use function, like so:
```js
  const start = Date.now();
  const sleepPromise: Promise<void> = sleep(expectedMs);

  // ...

  await sleepPromise;
  const actualMs = Date.now() - start;
```

Reply:
That's very much *less* straightforward, not more – it mixes the two control paths, and in subtly fragile ways.

In particular, `sleepPromise` will hang unless you advance the timer before `await`ing on it, and doing so would make the computation of `const actualMs` [0] meaningless.